Activities/Database/UiPath.Database.Activities/DatabaseConnect.cs

```diff
@@ -40,19 +41,33 @@ internal DatabaseConnect(IDBConnectionFactory factory)
 
         protected override IAsyncResult BeginExecute(AsyncCodeActivityContext context, AsyncCallback callback, object state)
         {
-            var connString = ConnectionString.Get(context);
-            var provName = ProviderName.Get(context);
-            Func<DatabaseConnection> action = () => _connectionFactory.Create(connString, provName);
-            context.UserState = action;
+            try
+            {
+                var connString = ConnectionString.Get(context);
+                var provName = ProviderName.Get(context);
+                Func<DatabaseConnection> action = () => _connectionFactory.Create(connString, provName);
+                context.UserState = action;
 
-            return action.BeginInvoke(callback, state);
+                return action.BeginInvoke(callback, state);
+            }
+            catch (DbException ex)
+            {
+                throw new Exception("[Database driver error]: " + ex.Message + " " + ex?.InnerException?.Message, ex);
```

Comment:
possible breaking change for a workflow that catches DbException. We should not change exception types. And if we throw we should be a bit more specific than Exception